### DISCUSSION TITLE
Release 0.3.5 armv6 and armhf

### DISCUSSION THREAD

**Discussion by solax76** [Spotifyd/spotifyd#1207] [General]
Hello everyone, 
release 0.3.5 is not avaialble for armv6 and armhf, while it was for 0.3.4. Is it possible to have also this release?
Thank you

**Comment by eladyn** [comment]
Thanks for the reminder, I wanted to this soon after the release, but somehow forgot about it.
Since Ubuntu 18.04 is no longer supported in GitHub pipelines, the CD got stuck and those two versions couldn't be released. Unfortunately, getting off of 18.04 is not as easy as bumping the version (at least for armv6), so I'm currently trying to get it working on a newer version. I'll let you know how it goes.

For `armhf`, things should be working, and you should be able to download artefacts from [here](https://github.com/eladyn/spotifyd/releases/tag/v0.3.50). I would appreciate testing for these binaries!

**Reply by eladyn** [reply]
I [finally](https://github.com/eladyn/spotifyd/actions) got the CD to run successfully (the binaries are available [here](https://github.com/eladyn/spotifyd/releases/tag/v0.3.50)). It's only Ubuntu 20.04 rather than 22.04 for now, but that should be easy to change as new Debian versions come out that are compatible with newer Ubuntu versions. If you have the time / possibilities to test the binaries for armv6 / armhf, that would be greatly appreciated! Otherwise, I'll be adding them tomorrow to the official release downloads. [Edit: done]

---

Relatedly, I feel like with that being successful, I might finally be able to tackle https://github.com/Spotifyd/spotifyd/pull/992 in the near future with a higher chance of getting it to work.

**Reply by mariocavicchi** [reply]
Issue?
```
$ cat /etc/os-release 
PRETTY_NAME="Raspbian GNU/Linux 10 (buster)"
NAME="Raspbian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=raspbian
ID_LIKE=debian
HOME_URL="http://www.raspbian.org/"
SUPPORT_URL="http://www.raspbian.org/RaspbianForums"
BUG_REPORT_URL="http://www.raspbian.org/RaspbianBugs"
```
```
$ ldd spotifyd
./spotifyd: /lib/arm-linux-gnueabihf/libm.so.6: version `GLIBC_2.29' not found (required by ./spotifyd)
./spotifyd: /lib/arm-linux-gnueabihf/libc.so.6: version `GLIBC_2.29' not found (required by ./spotifyd)
	linux-vdso.so.1 (0x7eff6000)
	/usr/lib/arm-linux-gnueabihf/libarmmem-${PLATFORM}.so => /usr/lib/arm-linux-gnueabihf/libarmmem-v7l.so (0x76306000)
	libdbus-1.so.3 => /lib/arm-linux-gnueabihf/libdbus-1.so.3 (0x7629f000)
	libasound.so.2 => /usr/lib/arm-linux-gnueabihf/libasound.so.2 (0x761be000)
	libgcc_s.so.1 => /lib/arm-linux-gnueabihf/libgcc_s.so.1 (0x76191000)
	librt.so.1 => /lib/arm-linux-gnueabihf/librt.so.1 (0x7617a000)
	libpthread.so.0 => /lib/arm-linux-gnueabihf/libpthread.so.0 (0x76150000)
	libm.so.6 => /lib/arm-linux-gnueabihf/libm.so.6 (0x760ce000)
	libdl.so.2 => /lib/arm-linux-gnueabihf/libdl.so.2 (0x760bb000)
	libc.so.6 => /lib/arm-linux-gnueabihf/libc.so.6 (0x75f6e000)
	/lib/ld-linux-armhf.so.3 (0x76f14000)
	libsystemd.so.0 => /lib/arm-linux-gnueabihf/libsystemd.so.0 (0x75ed1000)
	liblzma.so.5 => /lib/arm-linux-gnueabihf/liblzma.so.5 (0x75ea0000)
	liblz4.so.1 => /usr/lib/arm-linux-gnueabihf/liblz4.so.1 (0x75e74000)
	libgcrypt.so.20 => /lib/arm-linux-gnueabihf/libgcrypt.so.20 (0x75d9a000)
	libgpg-error.so.0 => /lib/arm-linux-gnueabihf/libgpg-error.so.0 (0x75d70000)
```

**Reply by slondr** [reply]
@mariocavicchi looks like you need to be on Debian 11 at least for that glibc version. You might be able to compile it yourself though

**Reply by mariocavicchi** [reply]
If someone use my same os version:
[spotifyd-pi-buster.zip](https://github.com/Spotifyd/spotifyd/files/11399346/spotifyd-pi-buster.zip)

**Reply by solax76** [reply]
I confirm that armv6 release is working fine! I'm using raspbian bullseye on raspberry zero w.
My main issue was the disconnections, but I need more time to test it.
Thank you!

PS: Sorry for the late response!

**Reply by eladyn** [reply]
Good to hear, thanks for the feedback! Regarding the buster problem: As you can see [here](https://askubuntu.com/a/445496), due to having to use Ubuntu 20.04 or higher, we're somewhat forced to use dependencies and a glibc version that is only compatible with buster (or newer?), even though it still is [in LTS mode](https://wiki.debian.org/LTS). A possible solution might be to go for statically the binaries, but I imagine that to be quite difficult due to the many native dependencies we have.

**Comment by MV10** [comment]
> For armhf, things should be working, and you should be able to download artefacts from [here](https://github.com/eladyn/spotifyd/releases/tag/v0.3.50). I would appreciate testing for these binaries!

I've been using `armhf-full` on my 32-bit Pi4B running the latest Bullseye without any problems.

Are any of the releases ARM64? I have a different program I suspect will perform better on the 64-bit OS.

Ah, just noticed: https://github.com/Spotifyd/spotifyd/issues/1209

One more thing to learn...
